### PR TITLE
Create documentation template

### PR DESCRIPTION
**Pull request by A-Baji**
This PR is to introduce the new documentation template for datajoint open source.
https://a-baji.github.io/element-array-ephys/

### Notes
* All files related to the documentation are kept within the `docs/` directory
* The first 3 configuration in `mkdocs.yml` are repo dependent.
  * The `repo_name` convention will be `org/repo`
* A `gh-pages` branch is necessary for deployment
* `docker-compose-docs-serve.yaml` can be used for development
* `docstring_example.py` has been added as a docstring example and includes links for formatting guides.
  * `docstring_example.py` may be deleted when no longer necessary.
  * see `src/api.md` for docstring usage
* The appropriate logo must be placed in `src/assets`
  * make sure to update lines [9](https://github.com/datajoint/element-array-ephys/pull/84/files#diff-7618337a88ea47f9b778a21f0e17d43eab2aba4880c6be2d95a7bf95e4b8b2dfR9) and [10](https://github.com/datajoint/element-array-ephys/pull/84/files#diff-7618337a88ea47f9b778a21f0e17d43eab2aba4880c6be2d95a7bf95e4b8b2dfR10) of `mkdocs.yml` to the correct logo name
* The landing page can be set on line [36](https://github.com/datajoint/element-array-ephys/pull/84/files#diff-7618337a88ea47f9b778a21f0e17d43eab2aba4880c6be2d95a7bf95e4b8b2dfR10) of `mkdocs.yml`

### CODE REVIEW

File: .github/workflows/development.yaml [hunk 1]
```diff
@@ -3,145 +3,166 @@ on:
...
+  #       with:
+  #         name: image-${{env.PKG_NAME}}-${{env.PKG_VERSION}}-py3.8-alpine
+  #     - name: Fetch pip artifacts
+  #       uses: actions/download-artifact@v2
+  #       with:
+  #         name: pip-${{env.PKG_NAME}}-${{env.PKG_VERSION}}
+  #         path: dist
+  #     - name: Publish pip release
+  #       run: |
+  #         export HOST_UID=$(id -u)
```
Comment (dimitri-yatsenko) [inline .github/workflows/development.yaml:122]: Do not comment out code. If it's not necessary, then remove it.

Comment (A-Baji) [reply .github/workflows/development.yaml:122]: It is necessary for publishing. I'm just disabling the workflows related to publishing temporarily so I can deploy to github pages on my fork.

**Comment by A-Baji** [comment]
A `gh-pages` branch needs to be made before merging.

**Review by guzman-raphael — APPROVED** [review]
@A-Baji Excellent work! :muscle:

**Pull request merged**